v2/managedsettling.go

```diff
@@ -113,11 +129,15 @@ func (m *ManagedSettler) handleError(ctx context.Context, settler MessageSettler
 			Reason:             to.Ptr(handleErr.Error()),
 			PropertiesToModify: nil,
 		})
+		// this could be a special hook to have more control on deadlettering, but keeping it simple for now
```

Comment:
Do you think it's a good idea to deadletter any time we hit failure and have run out of retries? What if when we see that the message has reached its retry limit, we use the Abandon settlement?

Also, correct me if I'm wrong, but we're eventually going to provide a default set of settlers to interface with the queue directly, right? So that users don't have to write their own settlements? I forgot how exactly that plays into the current structure we have now

Reply:
abandon requeues the message.. so that would basically retry the message when we run out of retry.
the user can set a MaxDelivery on the queue or subscription lower than the MaxRetry option and get the behavior you describe.
this is opioninated, moving to the deadletter from the application so that we can log it/track it, as opposed to letting SB moving it to the DLQ, which is invisible from the application.

Reply:
that makes sense, that we should let the user track it and handle it at the application level